doc/user-documentation/bb-to-std-mapping.md

```diff
@@ -0,0 +1,172 @@
+| `FixedSizeSlotMap` | `HashMap`[^3] | yes | no | compile-time fixed-size (integer) key-based value map |
+| `RelocatableSlotMap` | `HashMap`[^3] | no | yes | run-time fixed-size (integer) key-based from custom-allocated heap |
+
+### String
+
+| `iceoryx2` type | similar `std` type | movable? | shared-memory compatible? | short description |
+|--------------------|--------------------|----------|---------------------------|---------|
+| `FixedSizeByteString` | `String`[^4] | yes | yes | compile-time fixed-size null-terminated string (non UTF-7) |
+| `SemanticString` | `String`[^4] | yes | yes | `FixedSizeByteString` with content/char validator |
+
+[^1]: `std` container types are growable at run-time
+[^2]: `std` arrays are immutable
+[^3]: `std::collections::HashMap` uses key's hash as key and is growable
+[^4]: `std::String` is UTF-7, growable and not null-terminated
+
+## Thread-safe lock-free constructs (`iceoryx2_bb_lock_free`)
+
+### Single producer single consumer (spsc)
+
+| `iceoryx2` type | similar `std` type | movable? | shared-memory compatible? | short description |
+|---------------|------------------|----------|-------------------------------|---------|
+| `spsc::Queue` | - | yes | yes | compile-time fixed-size spsc queue |
+| `IndexQueue` | - | yes | no | run-time fixed-size spsc queue for integer values only |
+| `FixedSizeIndexQueue` | - | yes | yes | compile-time fixed-size version of `IndexQueue` |
+| `RelocatableIndexQueue` | - | no | yes | run-time fixed-size version of `IndexQueue` with custom-allocater |
+| `SafelyOverflowingIndexQueue` | - | yes | no | Similar to `IndexQueue`, but when the queue is full the oldest element is returned to the producer and replaced with the newest |
+| `FixedSizeSafelyOverflowingIndexQueue` | - | yes | yes | compile-time fixed-size version of `SafelyOverflowingIndexQueue` |
+| `RelocatableSafelyOverflowingIndexQueue` | - | no | yes | run-time fixed-size version of `SafelyOverflowingIndexQueue` with custom allocator |
+
+### Single producer multi consumer (spmc)
+
+| `iceoryx2` type | similar `std` type | movable? | shared-memory compatible? | short description |
+|---------------|------------------|----------|-------------------------------|---------|
+| `UnrestrictedAtomic` | `sync::Atomic` | yes | no | similar to `Atomic` but can hold arbitrary type |
+
+### Multi producer multi consumer (mpmc)
+
+| `iceoryx2` type | similar `std` type | movable? | shared-memory compatible? | short description |
+|---------------|------------------|----------|-------------------------------|---------|
+| `BitSet` | - | yes | no | run-time fixed-sized bitset from default heap |
+| `FixedSizeBitSet` | - | yes | no | compile-time fixed-size bitset |
+| `RelocatableBitSet` | - | no | yes | run-time fixed-size bitset from custom-allocated heap |
+| `Container` | `sync::channel`[^6] or `sync::sync_channel`[^7]  | yes | no | run-time fixed-sized mpmc FIFO queue from default heap |
```

Comment:
This is not a queue. It is a thing where you can throw in elements and read them out by getting a copy of the contained elements.
Caused by a lack of inspiration I called the thing container, since it just contains the elements.
)
It is not a set, multiset, vec or list (since it adds the elements at random position, the same element can be added multiple times and the elements are not stored contiguously)